File: src/main/java/com/ola/poz/service/AppointmentService.java
package com.ola.poz.service;

import com.ola.poz.exception.NotFoundException;
import com.ola.poz.model.Appointment;
import com.ola.poz.repository.AppointmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class AppointmentService {

    @Autowired
    private AppointmentRepository appointmentRepository;

    public Appointment getById(Long id) {
        Optional<Appointment> appointment = appointmentRepository.findById(id);
        if (!appointment.isPresent()) {
            throw new NotFoundException(String.format("Appointment with id %s does not exist", id));
        }
        return appointment.get();
    }

    public Appointment create(Appointment appointment) {
        return appointmentRepository.save(appointment);
    }

    public Appointment update(Appointment update, Long id) {
        Appointment existingAppointment = getById(id);

        existingAppointment.setPatient(update.getPatient());
        existingAppointment.setDescription(update.getDescription());
        existingAppointment.setAppointmentDate(update.getAppointmentDate());
        existingAppointment.setExaminations(update.getExaminations());

        return appointmentRepository.save(existingAppointment);
    }

    public void delete(Long id) {
        if (!appointmentRepository.existsById(id)) {
            throw new NotFoundException(String.format("Appointment with id %s does not exist", id));
        }
        appointmentRepository.deleteById(id);
    }

}

File: src/main/resources/static/appointment/appointment.service.js
angular.module('appointment')
    .service('appointmentService', function ($resource) {
        var service = this;
        var resource = $resource('http://localhost:8082/appointments/:appointmentId', {}, {
            update: {
                method: 'PUT'
            }
        });

        service.query = function () {
            return resource.query().$promise;
        }

        service.create = function (appointment) {
            return resource.save(null, appointment).$promise;
        }

        service.get = function(id){
            return resource.get({appointmentId:id}).$promise;
        }

        service.update = function (appointment) {
            return resource.update({appointmentId:appointment.id}, appointment).$promise;
        }

        service.remove = function (id) {
            return resource.remove({appointmentId:id}).$promise;
        }

    });